src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -258,6 +258,7 @@ std::shared_ptr<rpc::GcsNodeInfo> GcsNodeManager::RemoveNode(
     // Remove from cluster resources.
     gcs_resource_manager_->OnNodeDead(node_id);
     resources_buffer_.erase(node_id);
+    node_resource_usages_.erase(node_id);
```

Comment:
@WangTaoTheTonic Why didn't we delete this before? was there anything we are missing here?

Reply:
Sorry for the missing! Thanks for the fix Alex!